internal/internal_task_pollers.go

```diff
@@ -1207,7 +1207,7 @@ func convertActivityResultToRespondRequest(
 	dataConverter converter.DataConverter,
 	failureConverter converter.FailureConverter,
 	namespace string,
-	cancelAllowed bool,
```

Comment:
I don't think you need to change this, you can just change how `cancelAllowed` is computed